test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/CpsPackageReferenceProjectTests.cs

```diff
@@ -3804,6 +3804,138 @@ public async Task GetTransitivePackagesAsync_WithTransitivePackageReferences_Ret
+                IVsProjectAdapter projectAdapter = Mock.Of<IVsProjectAdapter>();
+                CpsPackageReferenceProject project = CreateCpsPackageReferenceProject(projectName, projectFullPath, projectCache);
+
+                ProjectNames projectNames = GetTestProjectNames(projectFullPath, projectName);
+                PackageSpec packageSpec = GetPackageSpec(projectName, projectFullPath, "[2.0.0, )");
+
+                // Restore info
+                DependencyGraphSpec projectRestoreInfo = ProjectTestHelpers.GetDGSpecFromPackageSpecs(packageSpec);
+                projectCache.AddProjectRestoreInfo(projectNames, projectRestoreInfo, new List<IAssetsLogMessage>());
+                projectCache.AddProject(projectNames, projectAdapter, project).Should().BeTrue();
+
+                // Package directories
+                var sources = new List<PackageSource>();
+                var packagesDir = new DirectoryInfo(Path.Combine(rootDir.SolutionRoot, "globalPackages"));
+                var packageSource = new DirectoryInfo(rootDir.PackageSource);
+                packagesDir.Create();
+                packageSource.Create();
+                sources.Add(new PackageSource(packageSource.FullName));
+
+                var logger = new TestLogger();
+                var request = new TestRestoreRequest(packageSpec, sources, packagesDir.FullName, logger)
+                {
+                    LockFilePath = Path.Combine(rootDir.SolutionRoot, "project.assets.json")
+                };
+
+                await SimpleTestPackageUtility.CreateFullPackageAsync(packageSource.FullName, "packageB", "1.0.0");
+                await SimpleTestPackageUtility.CreateFullPackageAsync(
+                    packageSource.FullName,
+                    "packageA",
+                    "2.15.3",
+                    new Packaging.Core.PackageDependency[]
+                    {
+                        new Packaging.Core.PackageDependency("packageB", VersionRange.Parse("1.0.0"))
+                    });
+
+                var command = new RestoreCommand(request);
+                RestoreResult result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+
+                // Act
+                ProjectPackages packages = await project.GetInstalledAndTransitivePackagesAsync(useTransitiveOrigins, CancellationToken.None);
+
+                // Assert
+                Assert.True(result.Success);
+                Assert.NotEmpty(packages.InstalledPackages);
+                Assert.NotEmpty(packages.TransitivePackages);
+
+                if (useTransitiveOrigins)
+                {
+                    Assert.All(packages.TransitivePackages, pkg => Assert.NotEmpty(pkg.TransitiveOrigins));
+                }
+                else
+                {
+                    Assert.All(packages.TransitivePackages, pkg => Assert.Empty(pkg.TransitiveOrigins));
+                }
```

Comment:
We should avoid conditional branching in test code. Instead, make a second input parameter for what's expected. This test should also be validating what is in the collection of transitive origins rather than just checking for empty / not empty.

Reply:
Separated into two tests.